src/cargo/ops/cargo_rustc/context.rs

```diff
@@ -417,7 +416,8 @@ impl<'a, 'cfg> Context<'a, 'cfg> {
         // we don't want to link it up.
         if src_dir.ends_with("deps") {
             // Don't lift up library dependencies
-            if unit.pkg.package_id() != &self.current_package && !unit.target.is_bin() {
+            if self.ws.current_opt().map_or(false, |p| unit.pkg.package_id() != p.package_id())
```

Comment:
What's the correct way to check this?

`ws.members().find(|&p| p == unit.pkg).is_some()` causes test failures.